_rules/text-node-contrast-afw4f7.md

```diff
@@ -0,0 +1,366 @@
+---
+id: afw4f7
+name: Text node has minimum contrast
+rule_type: atomic
+
+description: |
+  This rule checks that the highest possible contrast of every text node with its background meets the minimal contrast requirement
+
+accessibility_requirements:
+  wcag20:1.4.3: # Contrast (Minimum)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+
+input_aspects:
+  - DOM Tree
+  - CSS Styling
+  - Language
+
+acknowledgements:
+  authors:
+    - Brian Bors
+    - Kasper Isager
+    - Wilco Fiers
+---
+
+## Applicability
+
+Any [visible](#visible) character in a [text node](https://dom.spec.whatwg.org/#text) that is a [child](https://dom.spec.whatwg.org/#concept-tree-child) (in the [flat tree](https://drafts.csswg.org/css-scoping/#flat-tree)) of an HTML element, except if the [text node](https://dom.spec.whatwg.org/#text) is a [descendant](https://dom.spec.whatwg.org/#concept-shadow-including-descendant) of an element that:
```

Comment:
"Any visible character in a text node" means that the expectation will apply to the character, not to the text node. I'm pretty sure that this is not what you intended, otherwise a lot of passed examples fail because some of the characters do not meet the expectation.

Suggestion: Any text node with visible characters, that is a child ...